crates/ruff/src/rules/pycodestyle/rules/invalid_escape_sequence.rs

```diff
@@ -77,9 +95,35 @@ pub(crate) fn invalid_escape_sequence(
 
         prev = Some(i);
 
-        let Some(next_char) = body[i + 1..].chars().next() else {
+        let next_char = match body[i + 1..].chars().next() {
+            Some(next_char) => next_char,
+            None if tok.is_f_string_middle() => {
+                // If we're at the end of a f-string middle token, the next character
+                // is actually emitted as a different token. For example,
+                //
+                // ```python
+                // f"\{1}"
+                // ```
+                //
+                // is lexed as `FStringMiddle('\\')` and `LBrace`, so we need to check
+                // the next character in the source file.
+                //
+                // Now, if we're at the end of the f-string itself, the lexer wouldn't
+                // have emitted the `FStringMiddle` token in the first place. For example,
+                //
+                // ```python
+                // f"foo\"
+                // ```
+                //
+                // Here, there won't be any `FStringMiddle` because it's an unterminated
+                // f-string.
+                let Some(next_char) = locator.after(tok_range.end()).chars().next() else {
+                    continue;
+                };
+                next_char
```

Comment:
Nit: The `let` else is a bit more complicated to read because I first read the exceptional case before the "normal" flow

What we want to express here is, take the `next_char` if any and otherwise return

```rust
if let Some(next_char) locator.after(tok_range.end()).chars().next() {
    next_char
} else {
    continue;
}
```